### PR TITLE
Bug 1948780: Fix progress reporting of edge case when Rsync pods exit within first 5 seconds

### PR DESCRIPTION
**Pull request by pranavgaikwad**
This PR fixes an issue with progress reporting of Rsync pods where if the Rsync process exited within first 5 seconds, the progress updates werent picked up.

### CODE REVIEW

**Review by shubham-pampattiwar — APPROVED** [review]
VISIACK, Thank you @pranavgaikwad for fixing this.

File: pkg/controller/directvolumemigration/rsync.go [hunk 1]
```diff
@@ -1187,19 +1189,24 @@ func (t *Task) hasAllProgressReportingCompleted() (bool, error) {
...
 			case operation.Failed:
 				t.Owner.Status.FailedPods = append(t.Owner.Status.FailedPods, podProgress)
-			case dvmp.Status.PodPhase == corev1.PodSucceeded, operation.Succeeded:
+			case dvmp.Status.PodPhase == corev1.PodSucceeded:
 				t.Owner.Status.SuccessfulPods = append(t.Owner.Status.SuccessfulPods, podProgress)
-			case dvmp.Status.PodPhase == corev1.PodPending:
+			case dvmp.Status.PodPhase == corev1.PodPending, !operation.Failed:
 				t.Owner.Status.PendingPods = append(t.Owner.Status.PendingPods, podProgress)
-				pendingPods = append(pendingPods, fmt.Sprintf("%s/%s", podProgress.Namespace, podProgress.Name))
+				if dvmp.Status.CreationTimestamp != nil {
```
Comment (alaypatel07) [inline pkg/controller/directvolumemigration/rsync.go:1196]: Is dvmp creation timestamp is an accurate measure of the pod pending?

Comment (pranavgaikwad) [reply pkg/controller/directvolumemigration/rsync.go:1196]: @alaypatel07 that's Pod's creationTimestamp : https://github.com/konveyor/mig-controller/blob/f97cd1e001b122d6fbc73b95620b6a27f64bacb1/pkg/apis/migration/v1alpha1/directvolumemigrationprogress_types.go#L76

File: pkg/controller/directvolumemigration/rsync.go [hunk 2]
```diff
@@ -1187,19 +1190,24 @@ func (t *Task) hasAllProgressReportingCompleted() (bool, error) {
...
+					if time.Now().UTC().Sub(dvmp.Status.CreationTimestamp.Time.UTC()) > PendingPodWarningTimeLimit {
+						pendingSinceTimeLimitPods = append(pendingSinceTimeLimitPods, fmt.Sprintf("%s/%s", podProgress.Namespace, podProgress.Name))
+					}
+				}
 			}
 		}
 	}
 
+	isCompleted := len(t.Owner.Status.SuccessfulPods)+len(t.Owner.Status.FailedPods) == len(t.Owner.Spec.PersistentVolumeClaims)
```
Comment (alaypatel07) [inline pkg/controller/directvolumemigration/rsync.go:1206]: +1

**Review by alaypatel07 — APPROVED** [review]

**Pull request merged**